app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/GetMessagesController.scala

```diff
@@ -41,15 +42,42 @@ class GetMessagesController @Inject()(
                                      )(implicit ec: ExecutionContext)
   extends BackendController(cc) {
 
-  def getMessagesForMovement(lrn: String, updatedSince: Option[String]): Action[AnyContent] = {
+  def getMessagesForMovement(movementId: String, updatedSince: Option[String]): Action[AnyContent] = {
     // todo: how we handle error here if for example MongoDb throws?
     authAction.async(parse.default) {
       implicit request: EnrolmentRequest[AnyContent] => {
+        Try(UUID.fromString(movementId)) match {
+          case Success(_) =>
 
-        movementService.getMatchingERN(lrn, request.erns.toList).flatMap {
-          case None => Future.successful(BadRequest(Json.toJson(ErrorResponse(dateTimeService.timestamp(), "Invalid LRN supplied for ERN", ""))))
-          case Some(ern) => workItemService.addWorkItemForErn(ern, fastMode = false)
-            getMessagesAsJson(lrn, ern, updatedSince)
+            Try(updatedSince.map(Instant.parse(_))).map { updatedSinceTime => {
+
+              val movement = movementService.getMovementById(movementId)
+
+              movement.map {
+                case Some(mvt) =>
+                  if (getErnsForMovement(mvt).intersect(request.erns).isEmpty)
+                    BadRequest(Json.toJson(ErrorResponse(
+                      dateTimeService.timestamp(),
+                      "Invalid MovementID supplied for ERN",
+                      ""
+                    ))) else {
+                    workItemService.addWorkItemForErn(mvt.consignorId, fastMode = false)
+
+                    Ok(Json.toJson(filterMessagesByTime(mvt.messages, updatedSinceTime)))
+                  }
+                case _ => NotFound("blah")
```

Comment:
I think this falls straight through into the "else" case of the getOrElse below (or maybe not?) but can we put a better message here please?